index.bs

```diff
@@ -9036,6 +9044,98 @@ The [=remote end steps=] given |session|, and |command parameters| are:
 
 </div>
 
+#### The input.setFiles Command ####  {#command-input-setFiles}
+
+The <dfn export for=commands>input.setFiles</dfn> command sets a given file
+input with a given set of file paths.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      input.SetFiles = (
+        method: "input.setFiles",
+        params: input.SetFilesParameters
+      )
+
+      input.SetFilesParameters = {
+        context: browsingContext.BrowsingContext,
+        files: [*text]
+        ? element: script.SharedReference,
+      }
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl">
+     EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for input.setFiles">
+
+The [=remote end steps=] given |session|, and |command parameters| are:
+
+1. Let |context id| be the value of the <code>context</code> field of
+   |command parameters|.
+
+1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+   with |context id|.
+
+1. Let |document| be |context|'s [=active document=].
+
+1. Let |environment settings| be the [=environment settings object=] whose
+   [=relevant global object=]'s <a>associated <code>Document</code></a> is
+   |document|.
+
+1. Let |realm| be |environment settings|'s [=realm execution context=]'s
+   Realm component.
+
+1. Let |reference| be the value of the <code>element</code> field of
+   |command parameters|.
+
+1. If |reference| is not null,
+
+   1. Let |element| be the result of [=trying=] to [=deserialize remote
+      reference=] with |reference|, |realm|, and |session|.
+
+   1. If |element| doesn't implement {{HTMLInputElement}} return [=error=] with
+      [=error code=] [=unable to set file input=].
+
+   1. If the [=input type=] of |element| doesn't match <code>"file"</code>
+      return [=error=] with [=error code=] [=unable to set file input=].
+
+   1. [=trying|Try=] [=show the picker|showing the picker=] with |element|.
```

Comment:
I am not sure we really need to trigger the picker when we set the files. Perhaps, we can setFiles on an input unconditionally no matter if the dialog was shown or not.

Reply:
Yes, agreed. We should close the picker if it's already showing when this is called, but if it's not showing we should simply set the input value directly.